test/test_invalid_headers.py

```diff
@@ -239,20 +241,33 @@ class TestFilter(object):
     ]
 
     hdr_validation_combos = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, True),
-            (True, False),
-            (False, True),
-            (False, False)
-        ]
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in (
+            itertools.product([True, False], repeat=3)
+        )
     ]
 
     hdr_validation_no_trailers = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, False),
-            (False, False)
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in [
+            (True, False, False),
+            (False, False, False),
+        ]
+    ]
```

Comment:
@sunu I added this because there were certain combinations (specifically `is_trailer==True`) that were no longer considered valid after #305. I don't think we *need* to test `is_response_header` here since these test headers will never be a valid response, however, I don't think it hurts to test `is_response_header==False` either. This test class is specifically for broad filter checks, so the more coverage the better, imo.